src/Components/Components/src/HotReload/HotReloadEnvironment.cs

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Components.HotReload
+{
+    internal class HotReloadEnvironment
+    {
+        public static readonly HotReloadEnvironment Instance = new(Environment.GetEnvironmentVariable("DOTNET_MODIFIABLE_ASSEMBLIES") == "debug");
```

Comment:
Is there any chance that a static constructor prevents the linker from removing the type? I would guess that static constructors may always have side effects and hence always have to be left in and will run.

It's probably not a big deal. I'll leave it up to you whether you think it's worth doing in a less convenient way!

Reply:
Actually I have no idea how smart the linker is about figuring out whether the static constructor would run or not.

Reply:
Static ctors are hard to trim away and guarantee correctness. I believe the only time a static ctor will be trimmed is if the whole type can be trimmed. Maybe also if it was `BeforeFieldInit` and all the static fields were removed?

@vitek-karas and @MichalStrehovsky would know the exact rules here.

Reply:
I think the rule is basically:
* If it's `BeforeFieldInit` then linker may trim it
  * In this case the `.cctor` is preserved only if there's a field preserved on the type
* Otherwise it will always keep it if it keeps the type

Reply:
I was able to do some "fun" substitutions.xml to trim references to this type. Here's what the trimmed results look like now:

![image](https://user-images.githubusercontent.com/174281/111519599-7fe8f000-8714-11eb-84e7-777a9b4d8a41.png)

Removing the interface and the context is difficult because `ComponentBase` implements it. If we were determined we could turn the context in to a `Func<bool>` that would eliminate the context type.